src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -0,0 +1,214 @@
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds =>
+            ImmutableArray.Create(PreferStreamAsyncMemoryOverloads.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() =>
+            WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            CancellationToken ct = context.CancellationToken;
+            SyntaxNode root = await doc.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+
+            if (!(root.FindNode(context.Span, getInnermostNodeForTie: true) is SyntaxNode node))
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            if (!(model.GetOperation(node, ct) is IInvocationOperation invocation))
+            {
+                return;
+            }
+
+            // Defensive check to ensure the fix is only attempted on one of the 4 specific undesired overloads
+            if (invocation.Arguments.Length < 3 || invocation.Arguments.Length > 4)
+            {
+                return;
+            }
+
+            IArgumentOperation? bufferOperation = GetArgumentByPositionOrName(invocation.Arguments, 0, "buffer", out bool isBufferNamed);
+            if (bufferOperation == null)
+            {
+                return;
+            }
+
+            IArgumentOperation? offsetOperation = GetArgumentByPositionOrName(invocation.Arguments, 1, "offset", out bool isOffsetNamed);
+            if (offsetOperation == null)
+            {
+                return;
+            }
+
+            IArgumentOperation? countOperation = GetArgumentByPositionOrName(invocation.Arguments, 2, "count", out bool isCountNamed);
+            if (countOperation == null)
+            {
+                return;
+            }
+
+            // No nullcheck for this, because there is an overload that may not contain it
+            IArgumentOperation? cancellationTokenOperation = GetArgumentByPositionOrName(invocation.Arguments, 3, "cancellationToken", out bool isCancellationTokenNamed);
+
+            string title = MicrosoftNetCoreAnalyzersResources.PreferStreamAsyncMemoryOverloadsTitle;
+
+            Task<Document> fixInvocation = FixInvocation(doc, root, invocation, invocation.TargetMethod.Name,
+                                                         bufferOperation.Value.Syntax, isBufferNamed,
+                                                         offsetOperation.Value.Syntax, isOffsetNamed,
+                                                         countOperation.Value.Syntax, isCountNamed,
+                                                         cancellationTokenOperation?.Value.Syntax, isCancellationTokenNamed);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
```

Comment:
Interesting - I would have excepted this to flag [RS0005: DoNotUseCodeActionCreateRuleId](https://github.com/dotnet/roslyn-analyzers/blob/62863ff47591ccd19a7c543dda8b402978063752/src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs#L11). I will investigate why that rule did not fire.

Meanwhile, switch to the pattern used here to ensure we get telemetry on this code action usage: https://github.com/dotnet/roslyn-analyzers/blob/79c2638f54b9e7ef0a3954cc838ec4c13ea63c1f/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ImplementSerializationConstructors.Fixer.cs#L102-L109

Reply:
Okay np.